app/services/flood_risk_engine/address_services/deserialize/ea_facade_to_address.rb

```diff
@@ -0,0 +1,85 @@
+# Specific converter - convert EA AddressFacade into other usable forms
+
+module FloodRiskEngine
+  module AddressServices
+    module Deserialize
+
+      class EaFacadeToAddress
+
+        def self.contains_addresses?(inbound)
+          return true if inbound && inbound.is_a?(Hash) && !inbound["results"].empty?
+
+          false
+        end
+
+        def selectable_address_data(inbound)
+          addresses = inbound["results"] if inbound && inbound.is_a?(Hash)
+
+          addresses ||= []
+
+          results = addresses.collect do |a|
+            {
+              moniker: a["address"],
+              uprn: a["uprn"]
+            }
+          end
+
+          results
+        end
+
+        # Builds collection of attributes in correct format to simply
```

Comment:
Personally, I think @woodpigeon 's version is more readable, but that's a style thing and not something that should stop this shipping.